lib/hedgelog/scrub_replacement.rb

```diff
@@ -3,15 +3,18 @@ class ScrubReplacement
     def initialize(key, replacement)
       @key = key
       @replacement = replacement
-      @match_regex = /("?)#{@key}("?[=:]\s*"?)(.+?)(["&,;\s]|$)/
+      @match_regex = /("?)(#{@key})("?)(=?>?:?\s?)("?)([a-zA-Z0-9]*)("?)/
     end
 
     def scrub_string(string)
       string.gsub!(@match_regex) do
-        start = Regexp.last_match[1]
-        eql = Regexp.last_match[2]
-        delim = Regexp.last_match[4]
-        "#{start}#{@key}#{eql}#{@replacement}#{delim}"
+        quote1 = Regexp.last_match[1]
+        key = Regexp.last_match[2]
+        quote2 = Regexp.last_match[3]
+        separator = Regexp.last_match[4]
+        quote3 = Regexp.last_match[5]
+        quote4 = Regexp.last_match[7]
+        "#{quote1}#{key}#{quote2}#{separator}#{quote3}#{@replacement}#{quote4}"
       end
     end
 
```

Comment:
With the test updates below, I am open to alternatives, but this should get things passing again:
```suggestion
    def initialize(key, replacement)
      @key = key
      @replacement = replacement
      @match_regex = /("?)#{@key}\1(=>|=|:)(\s*)("?)(.+?)\4(&|,|;|\s|$)/
    end

    def scrub_string(string)
      string.gsub!(@match_regex) do
        quote1 = Regexp.last_match[1]
        sep = Regexp.last_match[2]
        whitespace = Regexp.last_match[3]
        quote2 = Regexp.last_match[4]
        # secret = Regexp.last_match[5]
        rest = Regexp.last_match[6]
        "#{quote1}#{@key}#{quote1}#{sep}#{whitespace}#{quote2}#{@replacement}#{quote2}#{rest}"
      end
    end
```